src/StreamJsonRpc.Shared/Reflection/TargetMethod.cs

```diff
@@ -28,12 +32,43 @@ internal TargetMethod(JsonRpcMessage request, object target, JsonSerializer json
             this.target = target;
 
             var targetMethods = new Dictionary<MethodSignature, object[]>();
+            bool errorFound = false;
             for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
             {
-                bool matchFound = false;
-                foreach (MethodInfo method in t.GetDeclaredMethods(request.Method))
+                var matchFound = false;
+                var methodName = request.Method;
+                var methodAttribute = request.Method;
+
+                if (methodToAttributeMap != null)
+                {
+                    foreach (string methodKey in methodToAttributeMap.Keys)
+                    {
+                        if (methodToAttributeMap[methodKey].Any(a => string.Equals(a, request.Method, StringComparison.OrdinalIgnoreCase)))
```

Comment:
The problem with reversing the dictionary is that to we still need to traverse the whole map to detect method name conflicts.  We could address this by having 2 dictionaries, but I thought that was a bit overkill since look up here should be pretty quick.